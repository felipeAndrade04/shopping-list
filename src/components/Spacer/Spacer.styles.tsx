import styled from 'styled-components/native';
import { SpacerProps } from './Spacer.types';

export const Container = styled.View<SpacerProps>`
  width: ${(props) => props.dimesion}px;
  height: ${(props) => props.dimesion}px;
`;
