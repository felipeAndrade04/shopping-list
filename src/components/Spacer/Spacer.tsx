import React from 'react';
import { SpacerProps } from './Spacer.types';
import * as S from './Spacer.styles';

export function Spacer({ dimesion = 8 }: SpacerProps) {
  return <S.Container dimesion={dimesion} />;
}
