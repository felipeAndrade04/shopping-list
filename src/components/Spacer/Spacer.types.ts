export interface SpacerProps {
  dimesion?: number;
}
