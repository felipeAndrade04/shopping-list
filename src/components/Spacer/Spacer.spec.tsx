import { render } from '@testing-library/react-native';
import { Spacer } from './Spacer';

describe('<Spacer />', () => {
  it('Should be render correctly', () => {
    const tree = render(<Spacer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
