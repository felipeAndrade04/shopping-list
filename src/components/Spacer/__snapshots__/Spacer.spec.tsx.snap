// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer /> Should be render correctly 1`] = `
<View
  dimesion={8}
  style={
    Array [
      Object {
        "height": 8,
        "width": 8,
      },
    ]
  }
/>
`;
